internal/sqlsmith: TestGenerateParse failed

[(internal/sqlsmith).TestGenerateParse failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2023067&tab=buildLog) on [master@a736f866c67728618363bae1e792306a2fa66877](https://github.com/cockroachdb/cockroach/commits/a736f866c67728618363bae1e792306a2fa66877):

```
I200618 22:40:53.315197 6924 sql/temporary_schema.go:595  [n1] temporary object cleaner next scheduled to run at 2020-06-18 23:10:53.31335139 +0000 UTC
I200618 22:40:53.318576 7039 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200618 22:40:53.321023 6923 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:38255 Attrs: Locality: ServerVersion:20.1-7 BuildTag:v20.2.0-alpha.1-479-ga736f866c6 StartedAt:1592520053245839562 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:40659} ClusterID:0acbd65d-181a-45c8-af6c-ca0876d2d7a4 StartedAt:1592520053245839562 LastUp:1592520053245839562}
I200618 22:40:53.334118 6430 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:20.1-7 User:root}
I200618 22:40:53.354221 7350 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:860a8e26-fdba-4e2c-99ad-45ff659d487a User:root}
I200618 22:40:53.361946 7426 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200618 22:40:53.367293 7447 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200618 22:40:53.406496 6922 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200618 22:40:53.418594 6724 server/server_sql.go:645  [n1] done ensuring all necessary migrations have run
I200618 22:40:53.418651 6724 server/server.go:1806  [n1] serving sql connections
I200618 22:40:53.418995 7618 sqlmigrations/migrations.go:727  [n1] starting wait for upgrade finalization before schema change job migration
I200618 22:40:53.419259 7618 sqlmigrations/migrations.go:759  [n1] starting schema change job migration
I200618 22:40:53.433262 7618 sqlmigrations/migrations.go:1177  [schema-change-job-migration] evaluating tables for creating jobs
I200618 22:40:53.433303 7618 sqlmigrations/migrations.go:779  [n1] schema change job migration completed
I200618 22:40:53.451322 7611 sql/event_log.go:132  [n1,client=127.0.0.1:60914,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:sql.stats.automatic_collection.enabled Value:false User:root}
I200618 22:40:53.456533 7611 sql/event_log.go:132  [n1,client=127.0.0.1:60914,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:sql.stats.histogram_collection.enabled Value:false User:root}
I200618 22:40:53.460414 7611 sql/event_log.go:132  [n1,client=127.0.0.1:60914,hostssl,user=root] Event: "create_table", target: 52, info: {TableName:defaultdb.public.table1 Statement:CREATE TABLE table1 (col1_0 NAME NOT NULL, col1_1 INET, col1_2 STRING, col1_3 FLOAT8 NOT NULL, PRIMARY KEY (col1_0 ASC, col1_3 ASC), UNIQUE (col1_2 DESC, col1_3 DESC, col1_1, col1_0), UNIQUE (col1_2 DESC), UNIQUE (col1_3 ASC), INDEX (col1_3, col1_2 DESC), UNIQUE (col1_3 ASC) STORING (col1_2)) User:root}
I200618 22:40:53.463512 7841 kv/kvserver/replica_command.go:413  [n1,split,s1,r34/1:/{Table/38-Max}] initiating a split of this range at key /Table/52 [r35] (zone config)
I200618 22:40:53.504404 6724 util/stop/stopper.go:539  quiescing
I200618 22:40:53.504480 6860 kv/kvserver/queue.go:580  [n1,s1] rate limited in gossip update (split): node unavailable; try another peer
I200618 22:40:53.504539 6860 kv/kvserver/queue.go:580  [n1,s1] rate limited in gossip update (merge): node unavailable; try another peer
W200618 22:40:53.504626 7894 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: TestGenerateParse (0.28s)
    sqlsmith_test.go:70: seed: 7504711476112964882
    sqlsmith_test.go:82: setup: rand-tables
    sqlsmith_test.go:92: setting: no-mutations+rand [disable mutations]
    sqlsmith_test.go:94: 
        		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
        		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
        		SET experimental_enable_enums = true;
        	CREATE TABLE table1 (col1_0 NAME NOT NULL, col1_1 INET, col1_2 STRING, col1_3 FLOAT8 NOT NULL, PRIMARY KEY (col1_0 ASC, col1_3 ASC), UNIQUE (col1_2 DESC, col1_3 DESC, col1_1, col1_0), UNIQUE (col1_2 DESC), UNIQUE (col1_3 ASC), INDEX (col1_3, col1_2 DESC), UNIQUE (col1_3 ASC) STORING (col1_2));
        ALTER TABLE table1 INJECT STATISTICS e'[{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_2"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"STRING","histo_buckets":[{"num_eq":2670369382725079786,"num_range":0,"distinct_range":0,"upper_bound":""},{"num_eq":0,"num_range":500,"distinct_range":0,"upper_bound":" 05 P"},{"num_eq":30000000000,"num_range":0,"distinct_range":0,"upper_bound":" s\\u0013"},{"num_eq":7000000,"num_range":2931740576479460406,"distinct_range":0,"upper_bound":"\\u003cB\\u000c:\\r^d\\u0013"},{"num_eq":8000000,"num_range":80,"distinct_range":80,"upper_bound":"A\\rD\\u0003"},{"num_eq":3000000,"num_range":80000000,"distinct_range":80000000,"upper_bound":"VMs\\u0018"},{"num_eq":5019877070381801775,"num_range":60000000,"distinct_range":0,"upper_bound":"W4\\u0003f"},{"num_eq":0,"num_range":2236780675672621098,"distinct_range":1756361847744899600,"upper_bound":"\\\\S"},{"num_eq":1451584495740850503,"num_range":40000000000,"distinct_range":34656088478.477585,"upper_bound":"ux\\u0013*M|"}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_3"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"FLOAT8","histo_buckets":[{"num_eq":2407408612067739334,"num_range":0,"distinct_range":0,"upper_bound":"-1.5690251852512311"},{"num_eq":1094070109891627014,"num_range":9000000000,"distinct_range":0,"upper_bound":"-0.6540603324115918"}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_0"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"NAME","histo_buckets":[{"num_eq":600000000,"num_range":0,"distinct_range":0,"upper_bound":""},{"num_eq":4937220280018928360,"num_range":400,"distinct_range":400,"upper_bound":"\\u0003"},{"num_eq":1130993885476782723,"num_range":3000000000,"distinct_range":0,"upper_bound":"\\u0006_"},{"num_eq":400,"num_range":967091894937502064,"distinct_range":967091894937502100,"upper_bound":"(~\\u001emgaiO("},{"num_eq":30000000000,"num_range":6039029440740798719,"distinct_range":4603416930929475000,"upper_bound":";F9o"},{"num_eq":600000,"num_range":600000,"distinct_range":600000,"upper_bound":"\\u003e~{D\\u000c\\t"},{"num_eq":10000,"num_range":3470332449049416548,"distinct_range":3470332449049416700,"upper_bound":"Wi\\u0014"},{"num_eq":7475455118443558146,"num_range":8394728098377859506,"distinct_range":1048322368212762600,"upper_bound":"|M{o+["}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_1"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":""}]';
        CREATE TYPE rand_typ_0 AS ENUM ('s');
        CREATE TYPE rand_typ_1 AS ENUM ('efmln', 'dx', 'eje', 'lhcj');
        CREATE TYPE rand_typ_2 AS ENUM ('cgv');
        CREATE TYPE rand_typ_3 AS ENUM ('va', 'f');
        CREATE TYPE rand_typ_4 AS ENUM ('s', 'dgs', 'sn', 'aspvzl', 's');
        
    sqlsmith_test.go:95: error executing '
        		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
        		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
        		SET experimental_enable_enums = true;
        	CREATE TABLE table1 (col1_0 NAME NOT NULL, col1_1 INET, col1_2 STRING, col1_3 FLOAT8 NOT NULL, PRIMARY KEY (col1_0 ASC, col1_3 ASC), UNIQUE (col1_2 DESC, col1_3 DESC, col1_1, col1_0), UNIQUE (col1_2 DESC), UNIQUE (col1_3 ASC), INDEX (col1_3, col1_2 DESC), UNIQUE (col1_3 ASC) STORING (col1_2));
        ALTER TABLE table1 INJECT STATISTICS e'[{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_2"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"STRING","histo_buckets":[{"num_eq":2670369382725079786,"num_range":0,"distinct_range":0,"upper_bound":""},{"num_eq":0,"num_range":500,"distinct_range":0,"upper_bound":" 05 P"},{"num_eq":30000000000,"num_range":0,"distinct_range":0,"upper_bound":" s\\u0013"},{"num_eq":7000000,"num_range":2931740576479460406,"distinct_range":0,"upper_bound":"\\u003cB\\u000c:\\r^d\\u0013"},{"num_eq":8000000,"num_range":80,"distinct_range":80,"upper_bound":"A\\rD\\u0003"},{"num_eq":3000000,"num_range":80000000,"distinct_range":80000000,"upper_bound":"VMs\\u0018"},{"num_eq":5019877070381801775,"num_range":60000000,"distinct_range":0,"upper_bound":"W4\\u0003f"},{"num_eq":0,"num_range":2236780675672621098,"distinct_range":1756361847744899600,"upper_bound":"\\\\S"},{"num_eq":1451584495740850503,"num_range":40000000000,"distinct_range":34656088478.477585,"upper_bound":"ux\\u0013*M|"}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_3"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"FLOAT8","histo_buckets":[{"num_eq":2407408612067739334,"num_range":0,"distinct_range":0,"upper_bound":"-1.5690251852512311"},{"num_eq":1094070109891627014,"num_range":9000000000,"distinct_range":0,"upper_bound":"-0.6540603324115918"}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_0"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":"NAME","histo_buckets":[{"num_eq":600000000,"num_range":0,"distinct_range":0,"upper_bound":""},{"num_eq":4937220280018928360,"num_range":400,"distinct_range":400,"upper_bound":"\\u0003"},{"num_eq":1130993885476782723,"num_range":3000000000,"distinct_range":0,"upper_bound":"\\u0006_"},{"num_eq":400,"num_range":967091894937502064,"distinct_range":967091894937502100,"upper_bound":"(~\\u001emgaiO("},{"num_eq":30000000000,"num_range":6039029440740798719,"distinct_range":4603416930929475000,"upper_bound":";F9o"},{"num_eq":600000,"num_range":600000,"distinct_range":600000,"upper_bound":"\\u003e~{D\\u000c\\t"},{"num_eq":10000,"num_range":3470332449049416548,"distinct_range":3470332449049416700,"upper_bound":"Wi\\u0014"},{"num_eq":7475455118443558146,"num_range":8394728098377859506,"distinct_range":1048322368212762600,"upper_bound":"|M{o+["}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_1"],"row_count":0,"distinct_count":0,"null_count":0,"histo_col_type":""}]';
        CREATE TYPE rand_typ_0 AS ENUM ('s');
        CREATE TYPE rand_typ_1 AS ENUM ('efmln', 'dx', 'eje', 'lhcj');
        CREATE TYPE rand_typ_2 AS ENUM ('cgv');
        CREATE TYPE rand_typ_3 AS ENUM ('va', 'f');
        CREATE TYPE rand_typ_4 AS ENUM ('s', 'dgs', 'sn', 'aspvzl', 's');
        ': pq: enum definition contains duplicate value "s"

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestGenerateParse PKG=./pkg/internal/sqlsmith TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestGenerateParse.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/sqlbase/testutils.go
+++ b/pkg/sql/sqlbase/testutils.go
@@ -1562,8 +1562,15 @@ func RandString(rng *rand.Rand, length int, alphabet string) string {
 func RandCreateType(rng *rand.Rand, name, alphabet string) tree.Statement {
 	numLabels := rng.Intn(6) + 1
 	labels := make([]string, numLabels)
-	for i := 0; i < numLabels; i++ {
-		labels[i] = RandString(rng, rng.Intn(6)+1, alphabet)
+	labelsMap := make(map[string]struct{})
+	i := 0
+	for i < numLabels {
+		s := RandString(rng, rng.Intn(6)+1, alphabet)
+		if _, ok := labelsMap[s]; !ok {
+			labels[i] = s
+			labelsMap[s] = struct{}{}
+			i++
+		}
 	}
 	un, err := tree.NewUnresolvedObjectName(1, [3]string{name}, 0)
 	if err != nil {

